### PR TITLE
ipn-handler - catch undefined constant

### LINKED ISSUE #6315: Failed order when only PayPal Express defined under PHP 8.1
[11-Mar-2024 18:28:17 UTC] PHP Fatal error:  Uncaught Error: Undefined constant "MODULE_PAYMENT_PAYPAL_ORDER_STATUS_ID" in /home/account/domains/client/public_html/ipn_main_handler.php:360

Perhaps we should define this if !defined after inclusion of application_top.php?

### PR DESCRIPTION
**Pull request by drbyte**
Fixes #6315 - Failed order when PayPal Standard not installed, under PHP 8.1

### CODE REVIEW

File: ipn_main_handler.php [hunk 1]
```diff
@@ -1,517 +1,584 @@
...
+                    //payment_status=Denied or failed
+                    $new_status = ($isECtransaction ? MODULE_PAYMENT_PAYPALWPP_REFUNDED_STATUS_ID : MODULE_PAYMENT_PAYPAL_REFUND_ORDER_STATUS_ID);
+                    break;
+                case 'echeck-cleared':
+                    $new_status = (int)MODULE_PAYMENT_PAYPAL_ORDER_STATUS_ID;
+                    break;
+                case ($txn_type === 'express-checkout-cleared' || str_starts_with($txn_type, 'cleared-')):
+                    //express-checkout-cleared
+                    $new_status = ($isECtransaction && defined('MODULE_PAYMENT_PAYPALWPP_ORDER_STATUS_ID') ? MODULE_PAYMENT_PAYPALWPP_ORDER_STATUS_ID : MODULE_PAYMENT_PAYPAL_ORDER_STATUS_ID);
+                    if (empty($new_status)) {
```
Comment (scottcwilson) [inline ipn_main_handler.php:527]: Should `MODULE_PAYMENT_PAYPAL_ORDER_STATUS_ID` in above statement be casted to int?

**Review by scottcwilson — COMMENTED** [review]
Looks good just one question.

File: ipn_main_handler.php (hunk 1 above)
Comment (lat9) [reply ipn_main_handler.php:527]: IMO, that entire conditional set into `$new_status` should be cast to int, not just that one constant.

Comment (drbyte) [reply ipn_main_handler.php:527]: Okay, will do that for consistency and for clarity to the human reader.

Everywhere the value gets used already first casts it to `int`. eg:
https://github.com/zencart/zencart/blob/445529df14235b84547b5c35f3b2d8341380c859/includes/functions/functions_osh_update.php#L85

Comment (drbyte) [reply ipn_main_handler.php:527]: Done.

**Review by scottcwilson — APPROVED** [review]

**Pull request merged**